code/common/os.q

```diff
@@ -7,10 +7,13 @@ ext:{`$(x?".")_(x:string x;x)[i:10h=type x]}
 del:{system("rm ";"del ")[NT],pth x}
 deldir:{system("rm -r ";"rd /s /q ")[NT],pth x}
 hdeldir:{[dirpath]
+ dirpath:$[10h=a:type dirpath;dirpath;-11h=a;string dirpath;'`type];
+ if[()~dirpath;:()];
```

Comment:
Won't the line above throw an error if it's the wrong type? Will the code ever get to this line and dirpath:()?

Reply:
Oh, and are we catching this error in your other code? and logging it? Or will this kill the function?